File: src/main/java/com/rdvdev2/timetravelmod/impl/ModSoundEvents.java
package com.rdvdev2.timetravelmod.impl;

import net.minecraft.sound.SoundEvent;
import net.minecraft.util.registry.Registry;

public class ModSoundEvents {

    public static final SoundEvent OLD_WEST_MUSIC = new SoundEvent(Mod.identifier("old_west_music"));

    public static void register() {
        registerSoundEvent("old_west_music", OLD_WEST_MUSIC);
    }

    public static void registerSoundEvent(String path, SoundEvent soundEvent) {
        Registry.register(Registry.SOUND_EVENT, Mod.identifier(path), soundEvent);
    }
}

File: src/main/kotlin/com/rdvdev2/timetravelmod/impl/ModSoundEvents.kt
package com.rdvdev2.timetravelmod.impl

import com.rdvdev2.timetravelmod.impl.Mod.identifier
import net.minecraft.sound.SoundEvent
import net.minecraft.util.registry.Registry

object ModSoundEvents {

    @JvmField
    val OLD_WEST_MUSIC = SoundEvent(identifier("old_west_music"))

    fun register() {
        OLD_WEST_MUSIC.registerAs("old_west_music")
    }

    private fun SoundEvent.registerAs(path: String) = Registry.register(Registry.SOUND_EVENT, identifier(path), this)
}